packages/swap/src/index.ts

```diff
@@ -113,9 +109,25 @@ class Swap extends EventEmitter {
     this.topTokenInfo = await fetch(TOP_TOKEN_INFO_LIST).then((res) =>
       res.json()
     );
-    this.providers = this.providerClasses.map(
-      (Provider) => new Provider(this.api, this.network)
-    );
+
+    // TODO: use network type instead?
+    switch (this.network) {
+      case SupportedNetworkName.Solana:
+        // Solana
+        this.providers = [new Jupiter(this.api as Web3Solana, this.network)];
+        break;
+      default:
+        // EVM
+        this.providers = [
+          new OneInch(this.api as Web3Eth, this.network),
+          new Paraswap(this.api as Web3Eth, this.network),
+          new Changelly(this.network),
+          new ZeroX(this.api as Web3Eth, this.network),
+          new Rango(this.api as Web3Eth, this.network),
```

Comment:
I believe Rango has solana support, we might have to edit their code to include it